clients/spark/core/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala

```diff
@@ -113,21 +192,67 @@ class LakeFSInputFormat extends InputFormat[Array[Byte], WithIdentifier[Entry]]
     val rangesReader: SSTableReader[RangeData] =
       SSTableReader.forMetaRange(job.getConfiguration, metaRangeURL)
     val ranges = read(rangesReader)
-    ranges.map(r =>
+    val res = ranges.map(r =>
       new GravelerSplit(
         r.message,
-        new Path(apiClient.getRangeURL(repoName, new String(r.id)))
+        new Path(apiClient.getRangeURL(repoName, new String(r.id))),
+        new String(r.id),
+        r.message.estimatedSize
       )
         // Scala / JRE not strong enough to handle List<FileSplit> as List<InputSplit>;
         // explicitly upcast to generate Seq[InputSplit].
         .asInstanceOf[InputSplit]
     )
+    logger.debug(s"Returning ${res.size} splits")
+    res
   }.asJava
+}
 
-  override def createRecordReader(
-      split: InputSplit,
-      context: TaskAttemptContext
-  ): RecordReader[Array[Byte], WithIdentifier[Entry]] = {
-    new EntryRecordReader(Entry.messageCompanion)
+class LakeFSAllRangesInputFormat extends LakeFSBaseInputFormat {
+  import LakeFSInputFormat._
+  override def getSplits(job: JobContext): java.util.List[InputSplit] = {
+    val conf = job.getConfiguration
+    val repoName = conf.get(LAKEFS_CONF_JOB_REPO_NAME_KEY)
+    var storageNamespace = conf.get(LAKEFS_CONF_JOB_STORAGE_NAMESPACE_KEY)
+    if (StringUtils.isBlank(repoName) && StringUtils.isBlank(storageNamespace)) {
+      throw new IllegalArgumentException(
+        "Must specify either LAKEFS_CONF_JOB_REPO_NAME_KEY or LAKEFS_CONF_JOB_STORAGE_NAMESPACE_KEY"
+      )
+    }
+    val apiClient = ApiClient.get(
+      APIConfigurations(
+        conf.get(LAKEFS_CONF_API_URL_KEY),
+        conf.get(LAKEFS_CONF_API_ACCESS_KEY_KEY),
+        conf.get(LAKEFS_CONF_API_SECRET_KEY_KEY),
+        conf.get(LAKEFS_CONF_API_CONNECTION_TIMEOUT_SEC_KEY),
+        conf.get(LAKEFS_CONF_API_READ_TIMEOUT_SEC_KEY)
+      )
+    )
+    if (StringUtils.isBlank(storageNamespace)) {
+      storageNamespace = apiClient.getStorageNamespace(repoName, StorageClientType.HadoopFS)
+    }
+    var nsURI = URI.create(storageNamespace)
+    nsURI = new URI(nsURI.getScheme,
+                    nsURI.getUserInfo,
+                    nsURI.getHost,
+                    nsURI.getPort,
+                    nsURI.getPath + (if (nsURI.getPath.endsWith("/")) "_lakefs" else "/_lakefs"),
+                    null,
+                    null
+                   )
+    val fs = FileSystem.get(nsURI, conf)
+    val it = fs.listFiles(new Path(nsURI.toString), false)
```

Comment:
Don't we need to scan just the ranges subprefix, and not the entire storage namespace?  (Or does that happen elsewhere?  If so, then `nsURI` is a really confusing name...)

Reply:
Bad name, sorry. Changing.